analysis/quantifier-comparisons/12-spatial-transcriptomics-storage.Rmd

```diff
@@ -0,0 +1,221 @@
+
+First we are testing use of a `SpatialExperiment` object to store the spatial transcriptomics results. 
+We will look at the Alevin-fry + Spaceranger quantification and use a custom function to create the combined object from both tools.
+
+```{r}
+# get path to fry output directory 
+fry_dir <- file.path(s3_out, "alevin-fry-knee", sample_id)
+fry_dir <- paste0(fry_dir, "-spliced_intron_txome_k31-salign-cr-like-em-knee")
+
+# paths to spatial folders 
+cellranger_folder <- paste0(sample_id, "-cdna-spatial")
+spatial_dir <- file.path(s3_out, "cellranger", cellranger_folder, "outs", "spatial")
+```
+
+```{r}
+# read in combined fry and spaceranger spe 
+fry_spe <- create_fry_spaceranger_spe(fry_dir, 
+                                       spatial_dir, 
+                                       sample_id)
+```
+
+### Importing as Seurat Object
+
+Below we will import the same data but as a `Seurat` object instead of a `SpatialExperiment` object. 
+
+```{r}
+# first create sce from Alevin output
+fry_sce <- scpcaTools::read_alevin(fry_dir,
+                                   usa_mode = TRUE,
+                                   which_counts = "spliced")
+
+# convert sce to seurat object 
+fry_seurat <- Seurat::CreateSeuratObject(counts = counts(fry_sce),
+                                 project = "SPATIAL",
+                                 assay = "Spatial")
+
+# read in image data using Seurat and subset to common cells
+image_data <- Seurat::Read10X_Image(spatial_dir)
+rownames(image_data@coordinates) <- gsub("-1","", rownames(image_data@coordinates))
+common_cells <- intersect(Seurat::Cells(x = fry_seurat), rownames(image_data@coordinates))
+image_data <- image_data[common_cells,]
+
+# add image data to Seurat
+Seurat::DefaultAssay(object = image_data) <- "Spatial"
+fry_seurat[['slice']] <- image_data
+```
+
+
+### Comparing Seurat and Spatial Experiment
+
+Now let's take a look at how each of these objects stores information for the experiment. 
+
+```{r}
+head(colData(fry_spe))
+head(fry_seurat@meta.data)
+```
+
+Seurat automatically adds some per cell QC metrics here, but in general the `colData` looks similar to what we would expect for a `SingleCellExperiment` for the `SpatialExperiment`.
+
+```{r}
+head(rowData(fry_spe))
+head(fry_seurat[["Spatial"]]@meta.features)
+```
+
+It looks like both of these start off with no `rowData` as expected.
+
+```{r}
+fry_spe <- fry_spe %>%
+  scuttle::addPerCellQCMetrics(subsets = list(mito_genes = mito_genes[mito_genes %in% rownames(fry_spe)])) 
+head(colData(fry_spe))
+```
+
+You can still use the same functions to add in per cell metrics that we are using in `scpcaTools` now on a `SpatialExperiment`. 
+
+Let's specifically look at how each of them stores the data related to the spatial information.
+The `SpatialExperiment` has three places where it stores spatial information, `spatialData`, `spatialCoords`, and `imgData`. 
+
+```{r}
+head(spatialData(fry_spe))
```

Comment:
Just a note that `spatialData` appears to be going away in a future version of this package. Nothing to do about it now, but worth keeping in mind. We might want to add a note/link here? https://github.com/drighelli/SpatialExperiment/issues/84